packages/headless/src/database/mongodb/branch/branch.model.ts

```diff
@@ -3,68 +3,71 @@ import { randomUUID } from 'crypto';
 import { Branch, BranchAddress } from 'src/entity/branch';
 
 export const BranchAddressSchema = new Schema<BranchAddress>(
-    {
-        id: {
-            type: String,
-            default: () => randomUUID(),
-            index: true,
-        },
-        addressLine1: String,
-        addressLine2: String,
-        postCode: String,
-        city: String,
-        country: String,
-    },
-    {
-        _id: false,
-        timestamps: false,
-        versionKey: false,
+  {
+    id: {
+      type: String,
+      default: () => randomUUID(),
+      index: true,
     },
+    addressLine1: String,
+    addressLine2: String,
+    postCode: String,
+    city: String,
+    country: String,
+  },
+  {
+    _id: false,
+    timestamps: false,
+    versionKey: false,
+  },
 );
 
 const BranchSchema = new Schema<Branch>(
-    {
-        id: {
-            type: String,
-            unique: true,
-            default: () => randomUUID(),
-        },
-        // storeId
-        store: {
-            type: String,
-            index: true
-        },
-        url: {
-            type: String,
-            unique: true
-        },
-        status: {
-            type: String,
-            enum: ['PENDING', 'ACCEPTED', 'REJECTED'],
+  {
+    id: {
+      type: String,
+      unique: true,
+      default: () => randomUUID(),
+    },
+    // storeId
+    store: {
+      type: String,
+      index: true,
+    },
+    url: {
+      type: String,
+      unique: true,
+    },
+    status: {
+      type: String,
+      enum: ['PENDING', 'APPROVED', 'REJECTED'],
+    },
+    address: BranchAddressSchema,
+    isActive: {
+      type: Boolean,
+      default: true,
+    },
+    image: {
+      cover: String,
+      logo: String,
+    },
+    inActiveReason: {
+      type: String,
+      enum: ['BLOCKED_BY_ADMIN', 'UNDER_MAINTENANCE', null],
+      default: null,
     },
-        address: BranchAddressSchema,
-        isActive: {
-            type: Boolean,
-            default: true,
-        },
-        image:{
-            cover: String,
-            logo: String
-        },
-        inActiveReason: {
-            type: String,
-            enum : ['BLOCKED_BY_ADMIN','UNDER_MAINTENANCE',null],
-            default: null
-        },
-        description:{
-            type: String,
-        },
-        name: String
+    description: {
+      type: String,
     },
-    {
-        timestamps: true,
-        versionKey: false,
+    name: {
+      type: String,
+      unique: true,
     },
```

Comment:
is this the branch name?
if so, then it can't be unique.
otherwise if you are modifying the original name then this will break the expectation of store owner. He will create a branch with a name, then he will find that the branch name has changed